### PR TITLE
[v2] more predictable slashdash

### PR DESCRIPTION
**Pull request by zkat**
Fixes: https://github.com/kdl-org/kdl/issues/401

### CODE REVIEW

**Comment by zkat** [comment]
also worth noting: This restricts `/- { child1 }` to only being present _after_ props and arguments, so the following is not legal:

```kdl
node /-{child1} foo;
```

but this is:

```kdl
node foo /-{one} {two} /-{three}
```

I believe this is what you were hoping for, @tabatkins?

File: SPEC.md [hunk 1]
```diff
@@ -764,25 +765,29 @@ document := bom? nodes
...
-optional-node-space := node-space*
+// Whitespace within nodes, where newline-ish things must be esclined.
+node-space := ws* escline ws* | ws+
 
-base-node := type? optional-node-space string (required-node-space node-prop-or-arg)* (required-node-space node-children)?
-node := base-node optional-node-space node-terminator
-final-node := base-node optional-node-space node-terminator?
-node-prop-or-arg := prop | value
+base-node := slashdash? type? node-space* string
+      (node-space+ node-prop-or-arg)*
```
Comment (tabatkins) [inline SPEC.md:777]: Just for consistency, I'd move the optional `slashdash` to here, rather than putting it into `node-prop-or-arg`.

**Review by tabatkins — APPROVED** [review]
Very minor style nit, otherwise r+

File: SPEC.md (hunk 1 above)
Comment (zkat) [reply SPEC.md:777]: Sounds good. Additionally, I just realized I didn’t put any space allowances around the /-. I’m thinking of making it like:

`slashdash := '/-' line-space*`

Which means that these would both be legal:

```kdl
// Very reasonable, IMO
/-
node 1 2 3

// interesting and I think acceptable—it means you can use them like esclines:

node 1 /-
2 3
```

We can totally ban the second behavior but I think it’s better than having to special case allowing esclines after the /-. This way, we just say “/- consumes all following line-space, and the next “item”, regardless of where you put it (as long as it’s a legal /- location“. 

Interestingly, that implies the following are legal too:

```kdl
node /- //next thing is commented
  foo

node /- /* commented */ foo
```

I’ll have to adjust the spec prose too, to clarify that these cases are possible.

**Pull request merged**